discovery-provider/src/tasks/index_network_peers.py

```diff
@@ -0,0 +1,174 @@
+    try:
+        identity_url = update_network_peers.shared_config['discprov']['identity_service_url']
+        identity_endpoint = urljoin(identity_url, 'registered_creator_nodes')
+
+        r = requests.get(identity_endpoint, timeout=3)
+        if r.status_code != 200:
+            raise Exception(f"Query to identity_endpoint failed with status code {r.status_code}")
+
+        registered_cnodes = r.json()
+        logger.info(f"Fetched registered creator nodes from chain via {identity_endpoint} - {registered_cnodes}")
+        return registered_cnodes
+    except Exception as e:
+        logger.error(f"Identity fetch failed {e}")
+        return []
+
+def refresh_cnodes_from_identity(self):
+    registered_cnodes = use_redis_cache(
+        'registered_cnodes_from_identity', 30, lambda: fetch_cnode_endpoints_from_identity(self)
+    )
+    return registered_cnodes
+
+
+# Function submitted to future in threadpool executor
+def connect_peer(endpoint, ipfs_client):
+    ipfs_swarm_address = get_ipfs_info_from_cnode_endpoint(endpoint, None)
+    ipfs_client.connect_peer(ipfs_swarm_address)
+
+# Actively connect to all peers in parallel
+def connect_peers(self, peers_list):
+    ipfs_client = update_network_peers.ipfs_client
+    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+        future_to_connect_peer_request = {
+            executor.submit(connect_peer, endpoint, ipfs_client): endpoint for endpoint in peers_list
+        }
+        for future in concurrent.futures.as_completed(future_to_connect_peer_request):
+            try:
+                # No return value expected here so we just ensure all futures are resolved
+                future.result()
+            except Exception as exc:
+                logger.error(exc)
+
+######## CELERY TASKS ########
+@celery.task(name="update_network_peers", bind=True)
+def update_network_peers(self):
+    # Cache custom task class properties
+    # Details regarding custom task context can be found in wiki
+    # Custom Task definition can be found in src/__init__.py
+    redis = update_network_peers.redis
+    ipfs_client = update_network_peers.ipfs_client
+    # Define lock acquired boolean
+    have_lock = False
+    # Define redis lock object
+    update_lock = redis.lock("update_network_peers", timeout=7200)
+    try:
+        # Attempt to acquire lock - do not block if unable to acquire
+        have_lock = update_lock.acquire(blocking=False)
+        if have_lock:
+            # An object returned from web3 chain queries
+            peers_from_ethereum = retrieve_peers_from_eth_contracts(self)
+            # An object returned from local database queries
+            peers_from_local = retrieve_peers_from_db(self)
+            # Combine the set of known peers from ethereum and within local database
+            all_peers = peers_from_ethereum
+            all_peers.update(peers_from_local)
+
+            # Legacy list of cnodes from identity
```

Comment:
interesting - was your intent in keeping this a fallback in case direct eth calls fail?
would be in favor of ripping it out, and also removing route from identity

Reply:
i think we should be careful from now on to _never_ ungracefully deprecate functionality in order to avoid unforeseen behavior that can result in user experience degradation

Reply:
once its clear that this has been working for some time we can then gracefully remove the legacy functionality and ensure zero UX degradation

Reply:
makes sense 👍